src/nvim/buffer_defs.h

```diff
@@ -781,6 +781,7 @@ struct file_buffer {
   signlist_T *b_signlist;       // list of signs to draw
   int b_signcols_max;           // cached maximum number of sign columns
   int b_signcols;               // last calculated number of sign columns
+  int b_signcols_cells;         // number of cells used for sign columns
```

Comment:
why do we need another field (and thus all the associated housekeeping)? why can't this replace `b_signcols`?

Reply:
count != cols, max/b_signcols holds the count.
https://github.com/blueyed/neovim/blob/6d5f8c7addcc4b276a436531e75cb3b620fe513e/src/nvim/buffer.c#L5163-L5167
But it certainly should at least use "cols" (effective).
I might be missing something here though.